core/lib/dal/src/transactions_dal.rs

```diff
@@ -524,19 +525,55 @@ impl TransactionsDal<'_, '_> {
             }
         }
 
-        transaction
-            .transactions_dal()
-            .handle_executed_l2_transactions(l2_block_number, block_base_fee_per_gas, transactions)
-            .await?;
-        transaction
-            .transactions_dal()
-            .handle_executed_l1_transactions(l2_block_number, transactions)
-            .await?;
-        transaction
-            .transactions_dal()
-            .handle_executed_upgrade_transactions(l2_block_number, transactions)
+        if insert_txs {
+            let tx_hashes: Vec<_> = transactions.iter().map(|tx| tx.hash.as_bytes()).collect();
+            sqlx::query!(
+                r#"
+                DELETE FROM transactions
+                WHERE
+                    hash = ANY ($1)
+                "#,
+                &tx_hashes as &[&[u8]],
+            )
```

Comment:
hmmmm how can this happen?

Reply:
Do you mean how EN can have txs that are not included in a block? If so then 2 cases:
1. recovery from main node postgres dump
2. block rollback